awswrangler/lakeformation/_utils.py

```diff
@@ -1,18 +1,134 @@
 """Utilities Module for Amazon Lake Formation."""
 import logging
 import time
-from typing import Any, Dict, List, Optional
+from typing import Any, Dict, List, Optional, Union
 
 import boto3
 
 from awswrangler import _utils, exceptions
+from awswrangler.catalog._utils import _catalog_id
+from awswrangler.s3._describe import describe_objects
 
 _QUERY_FINAL_STATES: List[str] = ["ERROR", "FINISHED"]
 _QUERY_WAIT_POLLING_DELAY: float = 2  # SECONDS
 
 _logger: logging.Logger = logging.getLogger(__name__)
 
 
+def _without_keys(d: Dict[str, Any], keys: List[str]) -> Dict[str, Any]:
+    return {x: d[x] for x in d if x not in keys}
+
+
+def _build_partition_predicate(
+    partition_cols: List[str],
+    partitions_types: Dict[str, str],
+    partitions_values: List[str],
+) -> str:
+    partition_predicates: List[str] = []
+    for col, val in zip(partition_cols, partitions_values):
+        if partitions_types[col].startswith(("tinyint", "smallint", "int", "bigint", "float", "double", "decimal")):
+            partition_predicates.append(f"{col}={str(val)}")
+        else:
+            partition_predicates.append(f"{col}='{str(val)}'")
+    return " AND ".join(partition_predicates)
+
+
+def _build_table_objects(
+    paths: List[str],
+    partitions_values: Dict[str, List[str]],
+    use_threads: bool,
+    boto3_session: Optional[boto3.Session],
+) -> List[Dict[str, Any]]:
+    table_objects: List[Dict[str, Any]] = []
+    paths_desc: Dict[str, Dict[str, Any]] = describe_objects(
+        path=paths, use_threads=use_threads, boto3_session=boto3_session
+    )
+    for path, path_desc in paths_desc.items():
+        table_object: Dict[str, Any] = {
+            "Uri": path,
+            "ETag": path_desc["ETag"],
+            "Size": path_desc["ContentLength"],
+        }
+        if partitions_values:
+            table_object["PartitionValues"] = partitions_values[f"{path.rsplit('/', 1)[0].rstrip('/')}/"]
+        table_objects.append(table_object)
+    return table_objects
+
+
+def _get_table_objects(
+    catalog_id: Optional[str],
+    database: str,
+    table: str,
+    transaction_id: str,
+    boto3_session: Optional[boto3.Session],
+    partition_cols: Optional[List[str]] = None,
+    partitions_types: Optional[Dict[str, str]] = None,
+    partitions_values: Optional[List[str]] = None,
+) -> List[Dict[str, Any]]:
+    """Get Governed Table Objects from Lake Formation Engine."""
+    session: boto3.Session = _utils.ensure_session(session=boto3_session)
+    client_lakeformation: boto3.client = _utils.client(service_name="lakeformation", session=session)
+
+    scan_kwargs: Dict[str, Union[str, int]] = _catalog_id(
+        catalog_id=catalog_id,
+        **{
```

Comment:
Why not simply pass it as regular args instead of unpack  a dictionary with static keys?

Reply:
Mostly because of the `PartitionPredicate` argument below which must only be passed when there is an actual partition predicate (it cannot be passed with None). Do you know of a better way to handle that?

Reply:
Oh the `scan_kwargs` seems perfect to me. I was talking about the `_catalog_id()` call receiving the arguments `TransactionId`, `DatabaseName`, `TableName` and  `MaxResults`. [Here](https://github.com/awslabs/aws-data-wrangler/blob/778170cee013e0030ed09190a5d5f6389b2999c7/awswrangler/catalog/_add.py#L32) we have an example about a similar call.

Reply:
Ahhh ok, yes you are right, now fixed!